src/scripts/main.js

```diff
@@ -1,3 +1,253 @@
 'use strict';
 
-// write your code here
+const message = document.querySelector('.message');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const button = document.querySelector('.button');
+const cells = document.querySelectorAll('.field-cell');
+const table = document.querySelector('.game-field');
+const score = document.querySelector('.game-score');
+
+const tableGame = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function render() {
+  for (let i = 0; i < tableGame.length; i++) {
+    const row = table.rows[i];
+
+    for (let j = 0; j < tableGame[i].length; j++) {
+      const cell = row.cells[j];
+
+      cell.className = 'field-cell';
+      cell.textContent = '';
+
+      if (tableGame[i][j] !== 0) {
+        cell.classList.add(`field-cell--${tableGame[i][j]}`);
+        cell.textContent = tableGame[i][j];
+      }
+    }
+  }
+}
+
+button.addEventListener('click', () => {
+  if (!button.classList.contains('restart')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'Restart';
+  }
+
+  for (let i = 0; i < tableGame.length; i++) {
+    for (let j = 0; j < tableGame[i].length; j++) {
+      tableGame[i][j] = 0;
+    }
+  }
+
+  message.classList.add('hidden');
+  score.textContent = 0;
+
+  cells.forEach((cell) => {
+    cell.className = 'field-cell';
+    cell.textContent = '';
+  });
+
+  addCell();
+  render();
+  button.blur();
+});
+
+document.addEventListener('keyup', (e) => {
+  if (
+    !button.classList.contains('restart')
+    || !messageLose.classList.contains('hidden')
+    || !messageWin.classList.contains('hidden')
+  ) {
+    return;
+  }
+
+  const copyTableGame = JSON.parse(JSON.stringify(tableGame));
+
+  switch (e.code) {
+    case 'ArrowLeft':
+      moveLeft();
+      break;
+
+    case 'ArrowRight':
+      moveRight();
+      break;
+
+    case 'ArrowUp':
+      moveUp();
+      break;
+
+    case 'ArrowDown':
+      moveDown();
+      break;
+
+    default:
+      return;
+  }
+
+  if (JSON.stringify(copyTableGame) === JSON.stringify(tableGame)) {
+    return;
+  }
+
+  addCell();
+  render();
+
+  if (
+    Array.from(cells).some((cell) =>
+      cell.classList.contains('field-cell--2048')
+    )
+  ) {
+    messageWin.classList.remove('hidden');
+  }
+
+  let endH = true;
+  let endV = true;
```

Comment:
Try to use more meaningful names for variables to improve the code readability and maintainability